### PR TITLE
Add postgresql-service subpackage

### PR DESCRIPTION
**Pull request by KavyaSree2610**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
To add postgresql-service as a sub-package

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES/NO**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/53282984

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=632704&view=results

### CODE REVIEW

File: SPECS/postgresql/postgresql.spec [hunk 1]
```diff
@@ -101,6 +118,50 @@ chown -Rv nobody .
...
+fi
+
+%post service
+PGDATA="/usr/local/pgsql/data"
+PGRUN="/run/postgresql"
+
+if [ ! -d "$PGDATA" ]; then
+    mkdir -p "$PGDATA"
+    chown postgres:postgres "$PGDATA"
+    su -c /usr/bin/initdb -D "$PGDATA" postgres
```
Comment (jslobodzian) [inline SPECS/postgresql/postgresql.spec:137]: I believe there was an issue with the -D option?  Is this understood/resolved?

File: SPECS/postgresql/postgresql.spec [hunk 2]
```diff
@@ -65,12 +71,21 @@ Obsoletes:      libpq-devel < 13
 The postgresql-devel package contains libraries and header files for
 developing applications that use postgresql.
 
+%package	service
+Summary:	To setup postgresql as a service
+Requires:	%{name} = %{version}-%{release}
+Requires:	util-linux
```
Comment (jslobodzian) [inline SPECS/postgresql/postgresql.spec:77]: Is the Requires needed at all times or just for "pre"-installation and "post" installation?
Requires:
    vs
Requires(pre):
Requires(post):

File: SPECS/postgresql/postgresql.spec [hunk 3]
```diff
@@ -65,12 +71,21 @@ Obsoletes:      libpq-devel < 13
 The postgresql-devel package contains libraries and header files for
 developing applications that use postgresql.
 
+%package	service
+Summary:	To setup postgresql as a service
+Requires:	%{name} = %{version}-%{release}
```
Comment (jslobodzian) [inline SPECS/postgresql/postgresql.spec:76]: Are you sure you don't need shadow-utils also?

**Review by mfrw — APPROVED** [review]
LGTM :)

**Pull request merged**